Samples/BackgroundBlur/BackgroundBlur/common/mfutils.h

```diff
@@ -0,0 +1,188 @@
+        IUnknown            *pSourceUnk = NULL;
+
+        // Create the source resolver.
+        if (SUCCEEDED(hr))
+        {
+            hr = MFCreateSourceResolver(&pSourceResolver);
+        }
+
+        // Use the source resolver to create the media source.
+        if (SUCCEEDED(hr))
+        {
+            MF_OBJECT_TYPE ObjectType = MF_OBJECT_INVALID;
+
+            hr = pSourceResolver->CreateObjectFromURL(
+                sURL,                       // URL of the source.
+                MF_RESOLUTION_MEDIASOURCE,  // Create a source object.
+                NULL,                       // Optional property store.
+                &ObjectType,                // Receives the created object type. 
+                &pSourceUnk                 // Receives a pointer to the media source.
+                );
+        }
+
+        // Get the IMFMediaSource interface from the media source.
+        if (SUCCEEDED(hr))
+        {
+            hr = pSourceUnk->QueryInterface(__uuidof(IMFMediaSource), (void**)ppSource);
+        }
+
+        // Clean up
+        SAFE_RELEASE(pSourceResolver);
+        SAFE_RELEASE(pSourceUnk);
+
+        return hr;
+    }
+
+    ///////////////////////////////////////////////////////////////////////
+    //  Name: GetStreamMajorType
+    //  Description:  Get the major media type from a stream descriptor.
+    //
+    //  Note: 
+    //  To get the major media type from a stream descriptor, you need to go
+    //  through the stream descriptor's media type handler. Use this helper
+    //  function if you don't need the type handler for anything else.
+    // 
+    /////////////////////////////////////////////////////////////////////////
+
+    inline HRESULT GetStreamMajorType(IMFStreamDescriptor *pSD, GUID *pguidMajorType)
+    {
+        CheckPointer(pSD, E_POINTER);
+        CheckPointer(pguidMajorType, E_POINTER);
+
+        HRESULT hr = S_OK;
+        IMFMediaTypeHandler *pHandler = NULL;
+
+        hr = pSD->GetMediaTypeHandler(&pHandler);
+        if (SUCCEEDED(hr))
+        {
+            hr = pHandler->GetMajorType(pguidMajorType);
+        }
+        SAFE_RELEASE(pHandler);
+        return hr;
+    }
+
+
+    //--------------------------------------------------------------------------------------
+    // Name: AllocGetWindowText
+    // Description: Helper function to get text from a window.
+    //
+    // This function allocates a buffer and returns it in pszText. The caller must
+    // call CoTaskMemFree on the buffer.
+    //
+    // hwnd:     Handle to the window
+    // pszText:  Receives a pointer to the string.
+    // pcchLen:  Receives the length of the string, in characters, not including
+    //           the terminating NULL character. This parameter can be NULL.
+    //--------------------------------------------------------------------------------------
+
+    inline HRESULT AllocGetWindowText(HWND hwnd, WCHAR **pszText, DWORD *pcchLen)
```

Comment:
This was duplicated in another file, the other file can probably call it from this header

Reply:
Removed mfutils.h from project because it's not used anywhere.